core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -143,13 +145,14 @@ class MetadataRequestTest extends BaseRequestTest {
     val response1 = sendMetadataRequest(new MetadataRequest.Builder(Seq(topic1, topic2).asJava, true, ApiKeys.METADATA.latestVersion).build())
     checkAutoCreatedTopic(topic1, topic2, response1)
 
-    // V3 doesn't support a configurable allowAutoTopicCreation, so the fact that we set it to `false` has no effect
-    val response2 = sendMetadataRequest(new MetadataRequest(requestData(List(topic2, topic3), false), 3.toShort))
-    checkAutoCreatedTopic(topic2, topic3, response2)
+    // V3 doesn't support a configurable allowAutoTopicCreation, so disabling auto-creation is not supported
+    intercept[UnsupportedVersionException] {
+      sendMetadataRequest(new MetadataRequest(requestData(List(topic2, topic3), false), 3.toShort))
+    }
```

Comment:
The original intent was for this to be ignorable. Have we decided that this is a bad idea?

Reply:
I wasn't sure, so left it as currently defined. Note that the MetadataRequest builder pretty much already makes it non-ignorable.

Reply:
Hmm.  I don't think allowAutoTopicCreation should be ignorable.  It seems like a relatively important request option that really changes the meaning of the request.

Reply:
Do we have a check for the case where `true` is used with older request types?

Reply:
Let me add something here.